### PR TITLE
Improve UX in selecting usernames to invite in parent org user invite modal

### PR DESCRIPTION
**Pull request by pavinduLakshan**
### Purpose

$subject

https://github.com/wso2/identity-apps/assets/41533942/b6310b42-44ff-4e28-b8bb-f733a6e58fbd

### Related Issues
- https://github.com/wso2/product-is/issues/19501#issuecomment-1938381086

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [X] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/users/components/guests/pages/invite-parent-org-user.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.invite-parent-org-user-form {
```
Comment (Achintha444) [inline apps/console/src/features/users/components/guests/pages/invite-parent-org-user.scss:1]: add license header

File: apps/console/src/features/users/components/guests/pages/invite-parent-org-user.tsx [hunk 2]
```diff
@@ -156,6 +159,32 @@ export const InviteParentOrgUser: FunctionComponent<InviteParentOrgUserPropsInte
                                     />
                                 ))
                             }
+                            filterOptions={ (_options: any, params: any) => {
```
Comment (Achintha444) [inline apps/console/src/features/users/components/guests/pages/invite-parent-org-user.tsx:162]: refrain from using `any` if possible

File: apps/console/src/features/users/components/guests/pages/invite-parent-org-user.scss (hunk 1 above)
Comment (pavinduLakshan) [reply apps/console/src/features/users/components/guests/pages/invite-parent-org-user.scss:19]: Thanks. fixed

File: apps/console/src/features/users/components/guests/pages/invite-parent-org-user.tsx (hunk 2 above)
Comment (pavinduLakshan) [reply apps/console/src/features/users/components/guests/pages/invite-parent-org-user.tsx:162]: Couldn't find a proper type for options and params. Therefore, kept it as it is. 

Anyways, updated the types in the `renderOption` method signature.

Comment (Achintha444) [reply apps/console/src/features/users/components/guests/pages/invite-parent-org-user.tsx:162]: ack

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**